src/main/java/org/isf/exatype/gui/ExamTypeEdit.java

```diff
@@ -222,49 +185,42 @@ private JButton getOkButton() {
 			okButton = new JButton();
 			okButton.setText(MessageBundle.getMessage("angal.common.ok"));  // Generated
 			okButton.setMnemonic(KeyEvent.VK_O);
-			okButton.addActionListener(new java.awt.event.ActionListener() {
-				public void actionPerformed(java.awt.event.ActionEvent e) {
-					ExamTypeBrowserManager manager = Context.getApplicationContext().getBean(ExamTypeBrowserManager.class);
-					try{
-						examType.setDescription(descriptionTextField.getText());
-						examType.setCode(codeTextField.getText());
-						
-						if (insert) {     // inserting
-							if (true == manager.newExamType(examType)) {
-	                           fireExamTypeInserted();
-	                           dispose();
-	                        } else
-	                        	JOptionPane.showMessageDialog(null,  MessageBundle.getMessage("angal.sql.thedatacouldnotbesaved"));
-	                    }
-	                    else {            // updating
-	                    	if (descriptionTextField.getText().equals(lastdescription)){
-	    						dispose();	
-	    					}else{
-								if (true == manager.updateExamType(examType)) {
-									fireExamTypeUpdated();
-									dispose();
-		                        }else
-		                        	JOptionPane.showMessageDialog(null, MessageBundle.getMessage("angal.sql.thedatacouldnotbesaved"));
-	    					}
+			okButton.addActionListener(e -> {
+				ExamTypeBrowserManager manager = Context.getApplicationContext().getBean(ExamTypeBrowserManager.class);
+				try {
+					examType.setDescription(descriptionTextField.getText());
+					examType.setCode(codeTextField.getText());
+
+					if (insert) {     // inserting
+						if (true == manager.newExamType(examType)) {
```

Comment:
`true ==` again and line 204 also

Reply:
it's from the old code moved to anonymous functions